### ISSUE TITLE
Unexpected ALL_OK on second $healthcheck request

### ISSUE THREAD

**Issue by prb112** [IBM/FHIR#1006]
**Description**
`$healthcheck` succeeds `ALL OK` on second request when first request fails and second request should fail. 

**To Reproduce**
1.  Use version 4.1.0 of the IBM FHIR Server
1. Configure a tenant TNT1 with a tenantKey.  The tenantKey should be invalid. 
1. Execute the following $healthcheck curl to get the failure. 

```sh
curl -k -v -X GET -u "fhiruser:*******" -H 'Content-Type: application/fhir+json'   'https://localhost:9443/fhir-server/api/v4/$healthcheck' -H 'X-FHIR-TENANT-ID: TNT1'
```

```json
{"resourceType":"OperationOutcome",
"id":"c0-a8-56-19-d2a6b730-86aa-44d1-912f-b0a69329e520",
"issue":[{"severity":"fatal","code":"exception","details":{"text":"FHIRPersistenceDBConnectException: Failure acquiring connection for datasource"}}]}
```

1. Execute the healthcheck for a second time with the same curl. 

```json
{"resourceType":"OperationOutcome","issue":[{"severity":"information","code":"informational","details":{"text":"All OK"}}]}
```
The All OK is unexpected, and should still return the failure. 

**Expected behavior**
The expected behavior is an error on each request. 

**Additional context**
Reported by @festewart

**Comment by prb112** [comment]
Verified the issue on 4.1.0
Debugged on 4.2.0, and found it is indirectly addressed.

**Comment by prb112** [comment]
Closing per confirmation fixed in 4.2.0

**Comment by prb112** [comment]
roughly this took a half day to step through and confirm
